src/fiskaltrust.Launcher/Commands/DoctorCommand.cs

```diff
@@ -0,0 +1,434 @@
+using fiskaltrust.storage.serialization.V0;
+using fiskaltrust.Launcher.Services.Interfaces;
+using Grpc.Net.Client;
+using ProtoBuf.Grpc.Client;
+using fiskaltrust.Middleware.Abstractions;
+using fiskaltrust.Launcher.AssemblyLoading;
+using fiskaltrust.Launcher.Clients;
+using fiskaltrust.ifPOS.v1.de;
+using fiskaltrust.ifPOS.v1;
+
+namespace fiskaltrust.Launcher.Commands
+{
+    public class DoctorCommand : RunCommand
+    {
+        public DoctorCommand(string name = "doctor") : base(name) { }
+    }
+
+    public class DoctorCommandHandler : ICommandHandler
+    {
+        public LauncherConfiguration ArgsLauncherConfiguration { get; set; } = null!;
+        public string LauncherConfigurationFile { get; set; } = null!;
+        public string LegacyConfigurationFile { get; set; } = null!;
+        public bool MergeLegacyConfigIfExists { get; set; }
+
+        private const string SUCCESS = "✅";
+        private const string ERROR = "❌";
+        private readonly ILifetime _lifetime;
+        private readonly LauncherExecutablePath _launcherExecutablePath;
+        private bool _failed = false;
+
+        public DoctorCommandHandler(ILifetime lifetime, LauncherExecutablePath launcherExecutablePath)
+        {
+            _lifetime = lifetime;
+            _launcherExecutablePath = launcherExecutablePath;
+        }
+
+        public async Task<int> InvokeAsync(InvocationContext context)
+        {
+            try
+            {
+                Log.Logger = new LoggerConfiguration()
+                    .AddLoggingConfiguration()
+                    .CreateLogger();
+
+                LauncherConfiguration launcherConfiguration = new();
+
+                if (File.Exists(LauncherConfigurationFile))
+                {
+                    launcherConfiguration = await CheckAwait("Parse launcher configuration", async () => LauncherConfiguration.Deserialize(await File.ReadAllTextAsync(LauncherConfigurationFile))) ?? new LauncherConfiguration();
+                }
+
+                if (MergeLegacyConfigIfExists && File.Exists(LegacyConfigurationFile))
+                {
+                    var legacyConfig = await CheckAwait("Parse legacy configuration file", async () => await LegacyConfigFileReader.ReadLegacyConfigFile(LegacyConfigurationFile));
+                    if (legacyConfig is not null)
+                    {
+                        launcherConfiguration.OverwriteWith(legacyConfig);
+                    }
+                }
+
+                launcherConfiguration.OverwriteWith(ArgsLauncherConfiguration);
+
+                launcherConfiguration.EnableDefaults();
+
+                var clientEcdh = await CheckAwait("Load ECDH Curve", async () => await CommonCommandHandler.LoadCurve(launcherConfiguration.AccessToken!, launcherConfiguration.UseOffline!.Value, dryRun: true), critical: false);
+                ftCashBoxConfiguration cashboxConfiguration = new();
+
+                if (clientEcdh is not null)
```

Comment:
Does it make sense to log a message when the curve cannot be loaded?

Reply:
The `CheckAwait` function above already prints an error when it failes. I've change it now so It prints warnings instead of errors when the `critical` parameter is set to false.